qutebrowser/mainwindow/tabwidget.py

```diff
@@ -181,6 +181,11 @@ def get_tab_fields(self, idx):
         except qtutils.QtValueError:
             fields['current_url'] = ''
 
+        try:
+            fields['protocol'] = self.tab_url(idx).scheme()
```

Comment:
Since we're calling `self.tab_url(idx)` a third time here already, can you please add a `url = self.tab_url(idx)` somewhere above (e.g. after all the `fields[...] = ...` assignments) and change the calls here to `url.host()` etc.?

Reply:
Will do!